packages/telemetry/src/block-slog.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-env node */
+
+import { makeFsStreamWriter } from '@agoric/internal/src/node/fs-stream.js';
+import {
+  makeContextualSlogProcessor,
+  SLOG_TYPES,
+} from './context-aware-slog.js';
+import getContextFilePersistenceUtils, {
+  DEFAULT_CONTEXT_FILE,
+} from './context-aware-slog-persistent-util.js';
+import { serializeSlogObj } from './serialize-slog-obj.js';
+
+/**
+ * @typedef {import('./context-aware-slog.js').Slog} Slog
+ * @typedef {ReturnType<ReturnType<typeof makeContextualSlogProcessor>>} ContextSlog
+ */
+
+/**
+ * @param {import('./index.js').MakeSlogSenderOptions} options
+ */
+export const makeSlogSender = async options => {
+  const { CHAIN_ID, CONTEXTUAL_BLOCK_SLOGS } = options.env || {};
+  if (!(options.stateDir || CONTEXTUAL_BLOCK_SLOGS))
```

Comment:
If both are needed, this condition isn't correct

```suggestion
  if (!options.stateDir || !CONTEXTUAL_BLOCK_SLOGS)
```

Reply:
No, atleast one of them is needed